torch_geometric/nn/conv/gat_conv.py

```diff
@@ -32,6 +37,120 @@
     from torch.jit import _overload_method as overload
 
 
+@torch.jit._overload
+def gat_norm(  # noqa: F811
+        edge_index, edge_weight, num_nodes, flow, dtype):
+    # type: (Tensor, Tensor, Optional[int], str, Optional[int]) -> Tuple[Tensor, Tensor]  # noqa
+    pass
+
+
+@torch.jit._overload
+def gat_norm(  # noqa: F811
+        edge_index, edge_weight, num_nodes, flow, dtype):
+    # type: (SparseTensor, Tensor, Optional[int], str, Optional[int]) -> Tuple[SparseTensor, Tensor]  # noqa
+    pass
+
+
+@torch.jit._overload
+def gat_norm(  # noqa: F811
+        edge_index, edge_weight, num_nodes, flow, dtype):
+    # type: (Adj, Tensor, Optional[int], str, Optional[int]) -> Tuple[Adj, Tensor]  # noqa
+    pass
+
+
+def gat_norm(  # noqa: F811
+    edge_index: Adj,
+    edge_weight: Tensor,
+    num_nodes: Optional[int] = None,
+    flow: str = "source_to_target",
+    dtype: Optional[torch.dtype] = None,
+):
+    fill_value = 1.0
+
+    if isinstance(edge_index, SparseTensor):
+        assert edge_index.size(0) == edge_index.size(1)
```

Comment:
Do we not support bipartite (N, M) case?  If not we should call it out in the documentation